File: solutions/Java/[0520_Easy] Detect Capital/detectCapitalUse.java
class Solution {
  public boolean isCapitalLetter(char c) {
    return Character.isUpperCase(c);
  }

  public boolean detectCapitalUse(String word) {
    int numberOfCapitalLetter = 0;
    int wordLength = word.length();

    for (int i = 0; i < wordLength; i++) {
      if (isCapitalLetter(word.charAt(i))) {
        numberOfCapitalLetter++;
      }
    }

    return (
      (isCapitalLetter(word.charAt(0)) && numberOfCapitalLetter == 1)
      || numberOfCapitalLetter == wordLength
      || numberOfCapitalLetter == 0
    );
  }
}

File: solutions/TypeScript/[0520_Easy] Detect Capital/detectCapitalUse.ts
function detectCapitalUse(word: string): boolean {
  const isCapitalLetter = (c: string) => (c.charCodeAt(0) >= 65 && c.charCodeAt(0) <= 90);

  let numberOfCapitalLetter = 0;
  const wordLength = word.length;

  for (let i = 0; i < wordLength; i++) {
    if (isCapitalLetter(word[i])) {
      numberOfCapitalLetter++;
    }
  }

  return (
    (isCapitalLetter(word[0]) && numberOfCapitalLetter === 1)
    || numberOfCapitalLetter === wordLength
    || numberOfCapitalLetter === 0
  );
}
